File: desert/decoy-drill.lua
function pos2xy(pos) return pos['x'], pos['y'] end

local decoysBuilt = 0
loop
    local item = self:findNearest(self:findItems())
    self:move(item.pos)
    if self.gold >= self:costOf("decoy") then
        self:buildXY("decoy", pos2xy(self.pos))
        decoysBuilt = decoysBuilt + 1
        if decoysBuilt >= 4 then
            break
        end
    end
end

self:say("Done building decoys!")
self:moveXY(14, 36)
self:say(decoysBuilt)

File: desert/decoy-drill.clj
(def decoysBuilt 0)
(while (< decoysBuilt 4)
  (def coin (.findNearest this (.findItems this)))
  (if coin
    (.move this (.pos coin))
    (.wait this 0.1))
  (if (>= (.gold this) (.costOf this "decoy"))
    (when true
      (.buildXY this "decoy" (.x (.pos this)) (.y (.pos this)))
      (def decoysBuilt (inc decoysBuilt)))))
(.say this "Done building decoys!")
(.moveXY this 14 36)
(.say this decoysBuilt)
